packages/order-utils/src/order_validation_utils.ts

```diff
@@ -146,12 +139,29 @@ export class OrderValidationUtils {
         expectedFillTakerTokenAmount?: BigNumber,
     ): Promise<void> {
         const orderHash = orderHashUtils.getOrderHashHex(signedOrder);
-        const filledTakerTokenAmount = await this._orderFilledCancelledFetcher.getFilledTakerAmountAsync(orderHash);
-        OrderValidationUtils._validateRemainingFillAmountNotZeroOrThrow(
-            signedOrder.takerAssetAmount,
-            filledTakerTokenAmount,
+        const isValidSignature = await signatureUtils.isValidSignatureAsync(
+            this._provider,
+            orderHash,
+            signedOrder.signature,
+            signedOrder.makerAddress,
         );
-        OrderValidationUtils._validateOrderNotExpiredOrThrow(signedOrder.expirationTimeSeconds);
+        if (!isValidSignature) {
+            throw new Error('INVALID_ORDER_SIGNATURE');
+        }
+
+        const isCancelled = await this._orderFilledCancelledFetcher.isOrderCancelledAsync(signedOrder);
+        if (isCancelled) {
+            throw new Error('CANCELLED');
```

Comment:
Since Exchange will revert with an `ORDER_UNFILLABLE` generic error we should refactor some of this to `getOrderInfo` and map the states to a enum of errors.

Happy for this to be a later TODO item, as it would be a larger refactor, replacing some of the balance/allowance cancel fetchers.

Reply:
@dekz in a later PR this entire class is going to be removed. Don't think it's worth making this change.